### PR TITLE
added component with tf slim model generator and pipeline adjustments

### PR DESCRIPTION
**Pull request by dtrawins**
New component generating slim models which can be used in combination with other components for demo and testing purposes.
In the PR is included a sample pipeline linking TF slim models generation with OpenVINO model optimization and converting to IR format.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/335)

### CODE REVIEW

**Comment by vicaire** [comment]
Would it be possible to add unit tests in later PRs? It may become difficult for developers to make changes to this doc otherwise.

File: contrib/components/openvino/tf-slim/containers/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# Similarly, we need to workaround sandboxing issues:
+#   https://github.com/bazelbuild/bazel/issues/418
+RUN echo "build --spawn_strategy=standalone --genrule_strategy=standalone" \
+    >>/etc/bazel.bazelrc
+# Install the most recent bazel release.
+ENV BAZEL_VERSION 0.15.0
+WORKDIR /
+RUN mkdir /bazel && \
+    cd /bazel && \
+    curl -H "User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36" -fSsL -O https://github.com/bazelbuild/bazel/releases/download/$BAZEL_VERSION/bazel-$BAZEL_VERSION-installer-linux-x86_64.sh && \
```
Comment (Ark-kun) [inline contrib/components/openvino/tf-slim/containers/Dockerfile]: Why does this need the User Agent header? There should be a comment about this.

File: contrib/components/openvino/tf-slim/containers/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+RUN cd tensorflow && bazel build tensorflow/tools/graph_transforms:summarize_graph
+
+FROM intelpython/intelpython3_core as PROD
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        git && \
+        apt-get clean
+
+WORKDIR /slim
+
+RUN git clone --depth 1 https://github.com/tensorflow/models && rm -Rf models/.git && \
```
Comment (Ark-kun) [inline contrib/components/openvino/tf-slim/containers/Dockerfile]: Using master is fragile. We should have an ARG and specify some release tag for it.

File: contrib/samples/openvino/model_optimizer/convert_model_pipeline.py [hunk 3]
```diff
@@ -15,12 +15,12 @@ def download_optimize_and_upload(
     dsl.ContainerOp(
         name='mo',
         image='<image name>',
-        command=['python3', '../convert_model.py'],
+        command=['convert_model.py'],
```
Comment (Ark-kun) [inline contrib/samples/openvino/model_optimizer/convert_model_pipeline.py]: Are you sure this is going to work without hashbang?

File: contrib/components/openvino/model_convert/containers/convert_model.py [hunk 4]
```diff
@@ -1,7 +1,9 @@
+#!/usr/bin/python3
```
Comment (Ark-kun) [inline contrib/components/openvino/model_convert/containers/convert_model.py]: #!/usr/bin/env python

File: contrib/components/openvino/model_convert/containers/convert_model.py [hunk 5]
```diff
@@ -77,19 +82,25 @@ def main():
...
     print("return code", return_code)
-    command = bin_path + "gsutil cp " + BIN + " " + args.output_path
+    command = "gsutil cp " + BIN + " " + os.path.join(args.output_path, os.path.split(BIN)[1])
     print("gsutil upload command", command)
     return_code = subprocess.call(command, shell=True)
     print("return code", return_code)
+    if return_code:
+        exit(1)
+
+    with open('/tmp/output_path.txt', 'w') as f:
```
Comment (Ark-kun) [inline contrib/components/openvino/model_convert/containers/convert_model.py]: /outputs/output_path/file

**Pull request merged**